pkg/kubelet/cm/container_manager.go

```diff
@@ -212,6 +224,30 @@ func int64Slice(in []int) []int64 {
 	return out
 }
 
+func podHasExclusiveCPUs(cr cpuAllocationReader, pod *v1.Pod) bool {
+	for _, container := range pod.Spec.InitContainers {
+		if containerHasExclusiveCPUs(cr, pod, &container) {
+			return true
+		}
+	}
+	for _, container := range pod.Spec.Containers {
+		if containerHasExclusiveCPUs(cr, pod, &container) {
+			return true
+		}
+	}
+	klog.V(4).InfoS("Pod contains no container with pinned cpus", "podName", pod.Name)
+	return false
+}
+
+func containerHasExclusiveCPUs(cr cpuAllocationReader, pod *v1.Pod, container *v1.Container) bool {
+	exclusiveCPUs := cr.GetExclusiveCPUs(string(pod.UID), container.Name)
+	if !exclusiveCPUs.IsEmpty() {
+		klog.V(4).InfoS("Container has pinned cpus", "podName", pod.Name, "containerName", container.Name)
```

Comment:
how did you pick log level of `4`? This is typically above the default so no logs will be produced.

Maybe not here, but ideally we need to have a log message with the default severity that will allow to detect that the logic worked (and suggest to disable it if it broke anything).

Reply:
Fair enough, let's review the logging levels to enable users troubleshooting

Reply:
kept specifics about container pinning at level `4`, added default (v=2) logging to convey when quota is deisabled